server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1347,4 +1363,42 @@ public void clearIndexShardCache(ShardId shardId, boolean queryCache, boolean fi
     public boolean isMetaDataField(String field) {
         return mapperRegistry.isMetaDataField(field);
     }
+
+    /**
+     * Checks to see if an operation can be performed without taking the cluster
+     * over the cluster-wide shard limit. Adds a deprecation warning or returns
+     * an error message as appropriate
```

Comment:
You don't have to wrap these so narrowly, we can use the full 140-column line length here.